cpp/src/structure/select_random_vertices_impl.hpp

```diff
@@ -105,11 +167,14 @@ rmm::device_uvector<vertex_t> select_random_vertices(
 
       std::vector<size_t> tx_value_counts(comm_size);
       for (int i = 0; i < comm_size; ++i) {
-        tx_value_counts[i] =
-          mg_sample_buffer.size() / comm_size +
-          (static_cast<size_t>(i) < static_cast<size_t>(mg_sample_buffer.size() % comm_size) ? 1
-                                                                                             : 0);
+        tx_value_counts[i] = mg_sample_buffer.size() / comm_size;
+      }
+
+      srand((unsigned)time(NULL));
```

Comment:
Do you know the resolution of std::time? If it is a second, it can be problematic if this function get called multiples times within a second.

Reply:
And better use `std::` for functions that belong to the standard library (e.g. srand & rand()).